views/partials/_navigation.njk

```diff
@@ -11,7 +11,7 @@
           <button hidden class="app-navigation__button js-app-navigation__button">
             {{ item.label }}
           </button>
-
+          <!-- [html-validate-disable-next aria-label-misuse --- inline fix for html-validate rule, rather than disable it entirely] -->
```

Comment:
Have we got an issue for this one?

Reply:
It was a deliberate choice for accessibility: https://github.com/alphagov/govuk-design-system/commit/4ff06e4961bcd4a147d580d1d5ac2736fe9f6ec3

@36degrees dya reckon I should make an issue to review this?

Reply:
I'm not sure where the list of 'allowed elements' on https://html-validate.org/rules/aria-label-misuse.html comes from… according to https://www.w3.org/TR/html-aria/#el-li `li` is not listed as ['Naming Prohibited'](https://www.w3.org/TR/html-aria/#dfn-naming-prohibited) and the [`listitem` role](https://www.w3.org/TR/wai-aria-1.2/#listitem) inherits the `aria-label` property… 🤔 

The rationale in the linked commit makes sense to me, but I think given it's been flagged it might be worth asking @davidc-gds to review it just in case? Whether or not we need an issue for that I'm unsure – maybe ask David?

Reply:
Interesting. Might dig about a bit in the `html-validate` source code and figure out if it's worth filing a bug. Will speak to David as well.

Reply:
Ah, so the `<ul>` text reads:

> [Global aria-* attributes](https://www.w3.org/TR/wai-aria-1.2/#global_states) and any aria-* attributes **applicable to the allowed roles**.

Which is sort of ambiguous, but I think basically means we need to apply a role to the `<ul>` element - if we do that, then `html-validate` is happy.

By default, the `<ul>` has the `list` role, which is:

> allowed, but NOT RECOMMENDED.

I guess the `menu` role feels most appropriate here?

A bit confusingly, in the case where this applies (when the mobile menu is visible), we automatically expand the "Community" sub-menu, but list the "Community" button as collapsed. It's arguably fine since it's just a set of links, but also maybe a bit weird.

Reply:
Basically, I think this could do with an issue 😂 

I'll make one

Reply:
Aah, of course – not sure why I was looking at `<li>`!

> By default, the `<ul>` has the `list` role, which is:
> 
> > allowed, but NOT RECOMMENDED.

I _believe_ when they say it's 'allowed but not recommended' they mean it's not recommended to set this role _explicitly_ – if you take a look at the headings for the table it says "Implicit ARIA semantics (explicitly assigning these in markup is NOT RECOMMENDED)".

> I guess the `menu` role or `navigation` feels most appropriate here? Though it's already contained within a `nav`.

We should avoid the navigation role for exactly that reason – it's already in a nav region. The [`menu` role](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/menu_role) is designed for interfaces that mimic desktop app style interfaces where you have menus with lists of items, like in a 'menu bar' and wouldn't be appropriate here.

`list` still feels like the appropriate role – I think if we changed it, we'd lose the information like 'List, 10 items' that screen reader users get which helps them understand how many links there are, for example.

Still worth having an issue to explore further, but I think either 1) having an `aria-label` is fine on a `<ul>` and is useful for users or 2) we should just remove the `aria-label`. I don't think changing the role makes sense.

Reply:
I've added an issue and linked to this discussion.

https://github.com/alphagov/govuk-design-system/issues/2624